extensions/big-data-cluster/src/extension.ts

```diff
@@ -14,9 +14,11 @@ import { AddControllerDialogModel, AddControllerDialog } from './bigDataCluster/
 import { ControllerNode } from './bigDataCluster/tree/controllerTreeNode';
 
 const localize = nls.loadMessageBundle();
+export let extensionMemento: vscode.Memento;
```

Comment:
Instead of exposing and accessing this directly I would suggest wrapping this in some sort of injectable API wrapper such as we do in other extensions like azurecore. That makes it much easier to unit test this (tests which are another thing that would be great to see. I know you have an issue out for adding tests already but even just adding one or two here would ensure that things are set up correctly to make the other tests easier to work with).

Reply:
fixed.